### PR TITLE
Refactors transit space reservation

### PR DESCRIPTION
**Pull request by Cyberboss**
Now anything can do it

@kevinz000

### CODE REVIEW

**Comment by kevinz000** [comment]
nice

**Comment by kevinz000** [comment]
#28908 is waiting on this one for what cyberboss told me to do, btw guys, just saying, its not stale.

**Comment by kevinz000** [comment]
reeeeeeeeee

File: code/controllers/subsystem/mapping.dm [hunk 1]
```diff
@@ -252,3 +256,96 @@ SUBSYSTEM_DEF(mapping)
...
+	unused_turfs = block(A, B)
+	var/datum/area_reservation/R = new
+	R.reserved_turfs = unused_turfs
+	qdel(R)	//this will prep the zone
+
+/datum/controller/subsystem/mapping/proc/RequestArea(width, height)
+	var/datum/area_reservation/R = new
+	if(!R.Reserve(width, height, unused_turfs, TRUE))
+		qdel(R)
+		return null
```
Comment (lzimann) [inline code/controllers/subsystem/mapping.dm]: just `return` will already return `null`, unless you really want it explicit for readability..?

File: code/controllers/subsystem/mapping.dm [hunk 2]
```diff
@@ -252,3 +256,96 @@ SUBSYSTEM_DEF(mapping)
...
+			proposed_zone = block(topleft, bottomright)
+			if(!proposed_zone)
+				continue
+			for(var/j in proposed_zone)
+				var/turf/T = j
+				if(!T)
+					continue base
+				if(!(T.flags & UNUSED_RESERVATION_TURF))
+					continue base
+			//to_chat(world, "[COORD(topleft)] and [COORD(bottomright)]")
```
Comment (optimumtact) [inline code/controllers/subsystem/mapping.dm]: screee

File: code/controllers/subsystem/mapping.dm [hunk 3]
```diff
@@ -252,3 +256,96 @@ SUBSYSTEM_DEF(mapping)
...
+*/
+
+/datum/controller/subsystem/mapping/proc/SetupReservedAreas()
+	// TODO: See if SHUTTLE_TRANSIT_BORDER is really necessary
+	var/turf/A = get_turf(locate(SHUTTLE_TRANSIT_BORDER, SHUTTLE_TRANSIT_BORDER, ZLEVEL_TRANSIT))
+	var/turf/B = get_turf(locate(world.maxx - SHUTTLE_TRANSIT_BORDER, world.maxy - SHUTTLE_TRANSIT_BORDER, ZLEVEL_TRANSIT))
+	unused_turfs = block(A, B)
+	var/datum/area_reservation/R = new
+	R.reserved_turfs = unused_turfs
+	qdel(R)	//this will prep the zone
```
Comment (optimumtact) [inline code/controllers/subsystem/mapping.dm]: wait what? why do you qdel it?

File: code/controllers/subsystem/mapping.dm [hunk 4]
```diff
@@ -252,3 +256,96 @@ SUBSYSTEM_DEF(mapping)
...
+	var/list/reserved_turfs	//list of all reserved_turfs
+	var/list/source_turfs	//list of turfs we took reserved_turfs from
+
+/datum/area_reservation/proc/Reserve(width, height, list/available_turfs, remove_source)
+	var/list/proposed_zone
+	var/turf/topleft
+	var/turf/bottomright
+	base:
+		for(var/i in available_turfs)
+			CHECK_TICK
```
Comment (optimumtact) [inline code/controllers/subsystem/mapping.dm]: isn't this done at roundstart? seems pointless to checktick

File: code/controllers/subsystem/mapping.dm (hunk 2 above)
Comment (optimumtact) [inline code/controllers/subsystem/mapping.dm]: remove it or put it behind an ifdef I suppose

File: code/controllers/subsystem/mapping.dm (hunk 3 above)
Comment (Cyberboss) [inline code/controllers/subsystem/mapping.dm]: The Destroy() will flag all the turfs as unused,

**Comment by Cyberboss** [comment]
This isn't working and I''m not sure why...

**Comment by lzimann** [comment]
@Cyberboss are you still doing this

File: code/controllers/subsystem/mapping.dm (hunk 3 above)
Comment (RemieRichards) [inline code/controllers/subsystem/mapping.dm]: this looks weird at first, but it's taking all of the turfs on ZLEVEL_TRANSIT where it makes logical sense to put a shuttle, and is flagging them all as ```UNUSED_RESERVATION_TURF``` (it also makes them space by default, which they should already be for ```ZLEVEL_TRANSIT``` but ho-hum)

**Comment by RemieRichards** [comment]
>not working

:/

**Comment by Cyberboss** [comment]
My sober self told me to ropen this

**Comment by Cyberboss** [comment]
You're drunk go home

**Comment by praisenarsie** [comment]
@Cyberboss finish thisssssssssssss

**Pull request closed without merging**